src/components/Form.vue

```diff
@@ -0,0 +1,268 @@
+<template>
+  <div id="container">
+    <button class="button" id="addUser" @click="showModal = true">
+      Add User
+    </button>
+    <transition name="fade" appear>
+      <div class="modal-overlay" v-if="showModal" @click="closeForm()"></div>
+    </transition>
+    <transition name="slide" appear>
+      <div class="modal" v-if="showModal">
+        <form class="form" id="form" v-on:submit.prevent="addNewUser">
+          <fieldset>
+            <legend>Candidate Info</legend>
+            <input
+              type="text"
+              name="field1"
+              placeholder="Enter Your Full Name *"
+              v-model="name"
+            />
+            <input
+              type="email"
+              name="field2"
+              placeholder="Enter Your Email Address *"
+              v-model="email"
+            />
+            <input type="date" name="field3" required v-model="dob" />
+            <input
+              type="password"
+              name="field4"
+              placeholder="Enter a password *"
+              v-model="password"
+            />
+            <input
+              type="password"
+              name="field5"
+              placeholder="Re-Enter the password *"
+              v-model="repassword"
+            />
+            <input type="checkbox" id="checkbox" name="field6" />
+            <label for="checkbox"
+              >Tick the checkbox if you agree to terms conditions *</label
+            >
+          </fieldset>
+          <input type="submit" value="Submit" />
+        </form>
+
+        <button class="button" @click="closeForm">
+          Close Form
+        </button>
+      </div>
+    </transition>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "Form",
+  props: ["users", "editUser", "action", "showModalProp", "editIndex"],
+  data() {
+    return {
+      showModal: false,
+      name: "",
+      email: "",
+      dob: "",
+      address: "",
+      password: "",
+      repassword: "",
+    };
+  },
+  watch: {
+    action(newVal) {
+      if (newVal === "edit") {
+        this.name = this.editUser.name;
+        this.email = this.editUser.email;
+        this.dob = this.editUser.dob;
+        this.showModal = !this.showModal;
+      } else {
+        this.name = "";
+        this.email = "";
+        this.dob = "";
+      }
+    },
+    showModalProp(newVal) {
+      this.showModal = newVal;
+    },
+  },
+
+  methods: {
+    addNewUser() {
+      if (this.name === "") {
+        return alert("Enter Name");
+      } else if (this.email === "") {
+        return alert("Enter Email");
+      } else if (this.dob === "") {
+        return alert("Enter Date of Birth");
+      } else if (this.password === "") {
+        return alert("Enter Password");
+      } else if (this.password != this.repassword) {
+        return alert("Both password should match exaclty");
+      } else if (document.getElementById("checkbox").checked == false) {
```

Comment:
don't use vanillay Js syntax use Vuejs syntax instead 
e.g., document.getElementById can be replaced by ref see vuejs docs